lib/src/solver/result.dart

```diff
@@ -19,6 +23,9 @@ class SolveResult {
   /// reachable from the root.
   final List<PackageId> packages;
 
+  /// The root package of this resolution.
+  final Package root;
```

Comment:
Why does a package resolution have a root package? is that reasonable...

I see now harm, and we do always create dummy root packages when we otherwise need to make a resolution. I'm just wondering if this is smart.

Reply:
Maybe, I should just be asking... why are we making this public?

Reply:
I managed to avoid this by passing an explicit root package.